src/main/java/it/gov/pagopa/controller/consent/PortalConsentControllerImpl.java

```diff
@@ -1,21 +1,17 @@
-package it.gov.pagopa.controller;
+package it.gov.pagopa.controller.consent;
 
 import com.fasterxml.jackson.core.JsonProcessingException;
 import it.gov.pagopa.dto.PortalConsentDTO;
 import it.gov.pagopa.service.PortalConsentService;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
-import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.RequestHeader;
-import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
-import java.util.*;
 
 
 @RestController
-//@RequestMapping("/idpay/portal/consent")
 public class PortalConsentControllerImpl implements PortalConsentController {
```

Comment:
it could be possible to centralize the NullResource map into 404 return code throwing a NullResourceException when desired (this way you have not more reason to build explicitly a ResponseEntity to set the header)
Thus you will write a @ControllerAdvice to intercept this exception and return a ResponseEntity configured with just the 404 response code.

I will let @StefanoTerragni to decide regarding this point

Reply:
I agree with you in order to organize all the logics which raise an Error to be captured by `@ControllerAdvice`